From f57c3c4b88bb39053c97c5c3a224fad29312c328 Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Wed, 25 Jul 2018 22:17:20 -0700
Subject: [PATCH 1596/1667] ASoC: SOF: update dai config after parsing pdm
 tokens

Update the dai config with the pdm config information after
parsing the pdm tokens. This will be needed to restore the dai
config after suspend/resume.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/topology.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index b3ff74d..44eae25 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -1710,6 +1710,9 @@ static int sof_link_dmic_load(struct snd_soc_component *scomp, int index,
 		dev_err(sdev->dev, "error: failed to set DAI config for DMIC%d\n",
 			config->dai_index);
 
+	/* update config with pdm config */
+	memcpy(config, ipc_config, sizeof(*ipc_config));
+
 	kfree(sdev->private);
 	kfree(ipc_config);
 
-- 
2.7.4

